src/main/kotlin/no/nav/tilleggsstonader/sak/utbetaling/iverksetting/IverksettDtoMapper.kt

```diff
@@ -0,0 +1,75 @@
+package no.nav.tilleggsstonader.sak.utbetaling.iverksetting
+
+import no.nav.tilleggsstonader.sak.behandling.domain.Saksbehandling
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.feilHvisIkke
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.Satstype
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.TilkjentYtelse
+import no.nav.tilleggsstonader.sak.utbetaling.tilkjentytelse.domain.TypeAndel
+import no.nav.tilleggsstonader.sak.vedtak.totrinnskontroll.domain.Totrinnskontroll
+import java.util.UUID
+
+object IverksettDtoMapper {
+
+    fun map(
+        behandling: Saksbehandling,
+        tilkjentYtelse: TilkjentYtelse,
+        totrinnskontroll: Totrinnskontroll,
+        iverksettingId: UUID,
+        forrigeIverksettingId: UUID?,
+    ): IverksettDto {
+        feilHvisIkke(tilkjentYtelse.andelerTilkjentYtelse.any { it.iverksetting?.iverksettingId == iverksettingId }) {
+            "Må inneholde en andel med iverksettingId=$iverksettingId"
+        }
+        return IverksettDto(
+            sakId = behandling.eksternFagsakId.toString(),
+            behandlingId = behandling.id,
+            iverksettingId = iverksettingId,
+            personident = behandling.ident,
+            forrigeIverksetting = behandling.forrigeBehandlingId?.let {
+                ForrigeIverksettingDto(
+                    it,
+                    forrigeIverksettingId ?: error("Mangler forrigeIverksettingId"),
+                )
+            },
```

Comment:
Håndteres det at forrige behandling er avslag (eller annet om ikke iverksettes)?

Reply:
https://github.com/navikt/tilleggsstonader-sak/pull/190/files#diff-46f2e118ed2282327c16fdf2c2f081dfc7dee31173a7b051b91e4e094808290fR57
Skal legge inn en test for det 👍

Reply:
Etter diskusjon: ForrigeBehandlingId er nødvendigvis ikke forrige behandling, men forrige behandlingId som ble iverksatt. Det betyder at det kan være iverksetting nr 2 for gitt behandling